flaml/searcher/blendsearch.py

```diff
@@ -267,25 +289,18 @@ def on_trial_complete(self, trial_id: str, result: Optional[Dict] = None,
             else:  # add to result cache
                 self._result[self._ls.config_signature(config)] = result
                 # update target metric if improved
-                objective = result[
-                    self._metric + self.lagrange] if self._metric_constraints \
-                    else result[self._metric]
+                objective = result[self._ls.metric]
                 if (objective - self._metric_target) * self._ls.metric_op < 0:
                     self._metric_target = objective
                 if not thread_id and metric_constraint_satisfied \
-                        and self._create_condition(result):
+                   and self._create_condition(result):
                     # thread creator
-                    self._search_thread_pool[self._thread_count] = SearchThread(
-                        self._ls.mode,
-                        self._ls.create(
-                            config, objective,
-                            cost=result.get(self.cost_attr, 1)),
-                        self.cost_attr
-                    )
                     thread_id = self._thread_count
-                    self._thread_count += 1
-                    self._update_admissible_region(
-                        config, self._ls_bound_min, self._ls_bound_max)
+                    self._started_from_given = self._candidate_start_points \
+                        and trial_id in self._candidate_start_points
+                    if not self._started_from_given:
+                        self._started_from_low_cost = True                    

```

Comment:
why don't we remove the trial_id from self._candidate_start_points here if a ls thread is created?